peps/pep-0783.rst

```diff
@@ -0,0 +1,189 @@
+is also increasingly being used by Python packages to provide interactive
+documentation.
+
+Pyodide currently maintains ports of 255 different packages at the time of this
+writing, including major scientific Python packages like NumPy, SciPy, pandas,
+Polars, scikit-learn, OpenCV, PyArrow, and Pillow as well as general purpose
+packages like aiohttp, Requests, Pydantic, cryptography, and orjson.
+
+About 60 packages are also testing against Pyodide in their CI, including NumPy,
+pandas, awkward-cpp, scikit-image, statsmodels, PyArrow, Hypothesis, and PyO3.
+
+Python package projects cannot deploy binary distributions for Pyodide on PyPI.
+Instead they must use other options like ``anaconda.org`` or ``jsdelivr.com``.
+This creates friction both for package maintainers and for users.
+
+
+Rationale
+=========
+
+Emscripten uses a variant of musl libc. The Emscripten compiler makes no ABI
+stability guarantees between versions. Many Emscripten updates are ABI
+compatible by chance, and the Rust Emscripten target behaves as if the ABI were
+stable with only `occasional negative consequences
+<https://github.com/rust-lang/rust/issues/131467>`__.
+
+There are several linker flags that adjust the Emscripten ABI, so Python
+packages built to run with Emscripten must make sure to match the ABI-sensitive
+linker flags used to compile the interpreter to avoid load-time or run-time
+errors. The Emscripten compiler continuously fixes bugs and adds support for new
+web platform features. Thus, there is significant benefit to being able to
+update the ABI.
+
+In order to balance the ABI stability needs of package maintainers with the ABI
+flexibility to allow the platform to move forward, Pyodide plans to adopt a new
+ABI for each feature release of Python.
+
+The Pyodide team also coordinates the ABI flags that Pyodide uses with the
+Emscripten ABI that Rust supports in order to ensure that we have support for
+the many popular Rust packages. Historically, most of the work for this has
+been related to unwinding ABIs. See for instance `this Rust Major Change
+Proposal <https://github.com/rust-lang/compiler-team/issues/801>`__.
+
+The ``pyodide`` platform tags only apply to Python interpreters compiled and
+linked with the same version of Emscripten as Pyodide, with the same
+ABI-sensitive flags.
+
+
+Specification
+=============
+
+The platform tags will take the form::
+
+    pyodide_${YEAR}_${PATCH}_wasm32
+
+Each one of these will be used with a specified Python version. For example, the
+platform tag ``pyodide_2025_0`` will be used with Python 3.13.
+
+Emscripten Wheel ABI
+--------------------
+
+The specification of the ``pyodide_<abi>`` platform includes:
+
+* Which version of the Emscripten compiler is used
+* What libraries are statically linked with the interpreter
+* What stack unwinding ABI is to be used
+* Which runtime platform features are required to be present
+
+and a handful of other similar details that affect the ABI.
```

Comment:
A PEP as a specification document should probably enumerate these 'handful' of other details?

Reply:
Addressed in 46837ec.